packages/monitors/src/OptimisticOracleContractMonitor.js

```diff
@@ -0,0 +1,198 @@
+// This module monitors OptimisticOracle contracts and produce logs when new price requests are submitted, proposed to, disputed, and settled.
+
+// TODO: Use pricefeed mapping from proposer bot to fetch approximate prices for each price request to give more log information. This probably means
+// refactoring the pricefeed mapping out of the proposer bot so it can be shared amongst clients.
+const { createEtherscanLinkMarkdown, createObjectFromDefaultProps, ZERO_ADDRESS } = require("@uma/common");
+
+class OptimisticOracleContractMonitor {
+  /**
+   * @notice Constructs new contract monitor module.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} optimisticOracleContractEventClient Client used to query OptimisticOracle Contract events for contract state updates.
+   * @param {Object} monitorConfig Monitor setting overrides such as log overrides.
+   * @param {Object} contractProps Configuration object used to inform logs of key contract information. Example:
+   *        E.g. { networkId:1 }
+   */
+  constructor({ logger, optimisticOracleContractEventClient, monitorConfig, contractProps }) {
+    this.logger = logger;
+
+    // Financial Contract event client to read latest contract events.
+    this.optimisticOracleContractEventClient = optimisticOracleContractEventClient;
+    this.optimisticOracleContract = this.optimisticOracleContractEventClient.optimisticOracleContract;
+    this.web3 = this.optimisticOracleContractEventClient.web3;
+
+    // Previous contract state used to check for new entries between calls.
+    this.lastRequestPriceBlockNumber = 0;
```

Comment:
Is the state stored with the client (looking at `latestEvents`)?

Reply:
Yep, so if monitor goes down it starts at 0. If running in serverless mode, then every iteration it begins at 0 because serverless "tears down" the bots after each iteration.